### PR TITLE
Initial commit for Babelfish v5.1

### PR DESCRIPTION
**Pull request by rishabhtanwar29**
### Description
This commit contains the following changes:
- Add version info and upgrade test for 14.15, 15.10, 16.6 and 17.2.
- Add upgrade schedule files for 14.16, 15.11, 16.7 and 17.3.
- Update T-SQL extension version to 5.1.0 and add following upgrade scripts:
  - babelfishpg_common--3.8.0--4.0.0.sql
  - babelfishpg_common--4.5.0--5.0.0.sql
  - babelfishpg_tsql--4.5.0--5.0.0.sql
  - babelfishpg_tsql--5.0.0--5.1.0.sql
- Update common extension version to 5.1.0 and add following upgrade scripts:
  - babelfish_common_helper--5.0.0--5.1.0.sql
  - spatial_types--5.0.0--5.1.0.sql

Task: NO-JIRA
Signed-off-by: Rishabh Tanwar <ritanwar@amazon.com>

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/configuration/upgrade-test-configuration.yml [hunk 1]
```diff
@@ -30,22 +30,6 @@ upgrade-version: [{
...
-    {
-      version: 14.6,
-      upgrade-type: minor
-    },
-    { 
-      version: target.latest,
-      upgrade-type: major
-    }
-  ]
-},
```
Comment (HarshLunagariya) [inline .github/configuration/upgrade-test-configuration.yml:48]: Why are we removing this?

File: .github/configuration/upgrade-test-configuration.yml [hunk 2]
```diff
@@ -149,7 +145,7 @@ upgrade-version: [{
 {
   upgrade-path: [
     {
-      version: 16.2,
+      version: 16.4,
       upgrade-type: null
```
Comment (HarshLunagariya) [inline .github/configuration/upgrade-test-configuration.yml]: Why this change?

File: contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--5.0.0--5.1.0.sql [hunk 3]
```diff
@@ -0,0 +1,46 @@
+-- complain if script is sourced in psql, rather than via ALTER EXTENSION
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--5.0.0--5.1.0.sql:1]: 4.5.0--5.0.0 TSQL upgrade script is required.

File: .github/configuration/upgrade-test-configuration.yml (hunk 1 above)
Comment (rishabhtanwar29) [reply .github/configuration/upgrade-test-configuration.yml:48]: Just trying to reduce the number of workflows here and since we already have 14.5 -> 14.6 -> latest, I think the above one seems unnecessary.

File: .github/configuration/upgrade-test-configuration.yml (hunk 2 above)
Comment (rishabhtanwar29) [reply .github/configuration/upgrade-test-configuration.yml]: We already have 16.1 -> latest so 16.2 seems unnecessary, so changed it to a newer version instead.

**Review by HarshLunagariya — APPROVED** [review]

**Review by jsudrik — APPROVED** [review]
Approved.
